MoreLinq.Test/CountDownTest.cs

```diff
@@ -0,0 +1,56 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2019 Atif Aziz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq.Test
+{
+    using System.Collections;
+    using System.Collections.Generic;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class CountDownTest
+    {
+        [Test]
+        public void IsLazy()
+        {
+            new BreakingSequence<object>()
+                .CountDown(42, BreakingFunc.Of<object, int?, object>());
+        }
+
+        static readonly IEnumerable<TestCaseData> Data =
```

Comment:
I used `TestCaseSource` because it gave 100% fidelity and flexibility in terms of using the most suitable types. Have a look at 09c5a1e8183552cca6984ccb8eb104d448304324 for the test refactored using test data in `[TestCase]`. Since you can't express nullable integers in attributes, you have to come up with a substitute like -1. You also cannot use tuples to express the combined expected result of the integer and countdown. Instead you have to code up the test in the test method body. I also attempted yet another refactoring using `ExpectedResult` in d322efb99df5210619032f81c4910c272b0cb50f but the problem of being able to express a tuple in attribute data remains the same. In the end I [reverted](https://github.com/morelinq/MoreLINQ/pull/461/commits/dc02138a631ceae00d84029f1eba0ae5a6238040) [both](https://github.com/morelinq/MoreLINQ/pull/461/commits/0ffc0b91de77014c0a2f053cbe38a5c8dc3a7168).